tests/upgrade/basic/task.yaml

```diff
@@ -12,6 +12,17 @@ restore: |
     rm -f /var/tmp/myevil.txt
     rm -f ./do-classic ./prevsnapdver.out
 
+    # an older version of snapd might have been missing proper context
+    # transitions for fontconfig cache locations, attempt to restore the context
+    # to avoid breaking tests that run after this one
```

Comment:
Isn't this a concern also for normal update in the distro?

Reply:
It is, but I would avoid fixing it programmatically in snapd as this may be overstepping distro boundaries a bit. Probably best if @Conan-Kudo could chime in. I fear, the situation regarding fontconfig cache in Fedora is not yet fully addressed.

Reply:
Yes, it's a concern sometimes, but if it is our fault, we might want a trigger in the package to restore the context during the package upgrade.

Reply:
I'll add a bit to the upgrade trigger.